src/main/java/store/service/PaymentService.java

```diff
@@ -0,0 +1,38 @@
+package store.service;
+
+import store.controller.CartManager;
+import store.domain.Cart;
+import store.domain.Receipt;
+
+public class PaymentService {
+    private static final double MEMBERSHIP_DISCOUNT_RATE = 0.3;
+    private final static int MEMBERSHIP_DISCOUNT_LIMIT = 8000;
+
+    private final CartManager cartManager;
+    private int membershipDiscountBalance;
+
+    public PaymentService(CartManager cartManager) {
+        this.cartManager = cartManager;
+        this.membershipDiscountBalance = MEMBERSHIP_DISCOUNT_LIMIT;
+    }
+
+    public Cart getCartItems(String purchaseList) {
+        return cartManager.generateCart(purchaseList);
+    }
+
+    public Receipt createReceipt(Cart cart, boolean applyMembership) {
+        int membershipDiscount = calcMembershipDiscount(applyMembership, cart.totalRetailPrice());
+        return new Receipt(cart, cart.totalPrice(), cart.totalFreePrice(), membershipDiscount);
+    }
+
+    private int calcMembershipDiscount(boolean apply, int eligibleAmount) {
+        if (apply) {
+            int discountAmount = Math.min((int) (eligibleAmount * MEMBERSHIP_DISCOUNT_RATE),
+                    membershipDiscountBalance);
```

Comment:
라인 포맷팅을 더 이쁘게 할 수 있을 것 같아요!
모듈화 하는 것도 방법인것 같아요😃
```suggestion
            int discountAmount = Math.min(
                    (int) (eligibleAmount * MEMBERSHIP_DISCOUNT_RATE),
                    membershipDiscountBalance
            );
```